lib/active_model/serializer/adapter/json.rb

```diff
@@ -4,44 +4,122 @@ class ActiveModel::Serializer::Adapter::Json < ActiveModel::Serializer::Adapter
 
         def serializable_hash(options = nil)
           options ||= {}
+          hash = {}
+
           if serializer.respond_to?(:each)
-            result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            hash = serialize_array_without_root(serializer, options)
           else
-            hash = {}
+            hash = resource_object_for(serializer)
+            hash = add_resource_relationships(hash, serializer)
+          end
+
+          { root => hash }
+        end
+
+        def fragment_cache(cached_hash, non_cached_hash)
+          ActiveModel::Serializer::Adapter::Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
```

Comment:
You could use the opportunity to remove the useless `()` after `FragmentCache.new`.